tests/epyccel/test_arrays_multiple_assignments.py

```diff
@@ -165,6 +166,69 @@ def f(c):
     c = np.random.random()
     assert f(c) == g(c)
 
+#==============================================================================
+def test_Reassign_to_Target(language):
+
+    def f():
+        import numpy as np
+        x = np.zeros((3, 7), dtype=int)
+        c = x
+        x = np.ones ((4, 5), dtype=int)
+        return c.sum()
+
+     # Initialize singleton that stores Pyccel errors
+    errors = Errors()
+
+    # epyccel should raise an Exception
+    with pytest.raises(PyccelSemanticError):
+        epyccel(f, language=language)
+
+    # Check that we got exactly 1 Pyccel error
+    assert errors.has_errors()
+    assert errors.has_warnings()
+    assert errors.num_messages() == 2
```

Comment:
The assertions that are carried out here do not match what you say you are doing in the comment. To check that you have exactly 1 Pyccel error I would expect:
```
    assert errors.has_errors() ?== 1
    assert errors.has_warnings() == 0
    assert errors.num_messages() == 1
```

Reply:
fixed